pkg/fs/unique.go

```diff
@@ -0,0 +1,33 @@
+package fs
+
+import (
+	"fmt"
+	"path/filepath"
+	"sort"
+	"strings"
+)
+
+// UniquePath determines a unique path given the desired name and parent
+// directory. For instanace of the desired path in /usr/mypath but such a path
+// already exists, the path /usr/mypath_01 is returned (and so on).
+// File extensions are respected and the unique number is appended before any
+// extensions (e.g. mypath.txt -> mypath_01.txt).
+func UniquePath(parent, name string) (string, error) {
+	ext := filepath.Ext(name)
+	base := strings.TrimSuffix(name, ext)
+	matches, err := filepath.Glob(filepath.Join(parent, base+"*"+ext))
```

Comment:
I don't think this does the right thing if `name` doesn't exist but `base+"x"+ext` does. I suggest doing something similar to what ioutil.TempDir does.

Reply:
ok